### ISSUE TITLE
After upgrade from 0.6.15 to 0.6.16 (docker) many 500 errors

### ISSUE THREAD

**Issue by ictcorebiz** [gogits/gogs#1765]
After I upgraded from docker image gogs/gogs:latest that contained 0.6.15 to the current image 0.6.16, I can no longer access either issues or repositories.

Situation:
1 user - no repository
1 organisation - 1 repository

<code>
docker run --restart always --detach --name=gogs -p 10022:22 -p 10080:3000 -v /data/docker/gogs:/data gogs/gogs
</code>

after pulling the new gogs docker image, stopping and removing the old container, and restarting the above docker run, all started normally.

I can login to the UI, I see all entries on the dashboard, but the moment I want to browse the "organisation" repository (or the issues), I get an error 500.

When doing a docker logs gogs, I get the following output:
<pre>
[Macaron] Started GET /organisation/website.git/info/refs?service=git-upload-pack for 172.x.x.x
[Macaron] Completed /organisation/website.git/info/refs?service=git-upload-pack 401 Unauthorized in 1.861422ms
[Macaron] Started GET /organisation/website.git/info/refs?service=git-upload-pack for 172.x.x.x
[Macaron] Completed /organisation/website.git/info/refs?service=git-upload-pack 401 Unauthorized in 1.81288ms
[Macaron] Started GET /organisation/website.git/info/refs?service=git-upload-pack for 172.x.x.x
[Macaron] Completed /organisation/website.git/info/refs?service=git-upload-pack 404 Not Found in 114.099148ms
[Macaron] Started GET / for 192.x.x.x
[Macaron] Completed / 200 OK in 27.144833ms
[Macaron] Started GET /Organisation/website for 192.x.x.x
2015/10/12 19:40:25 [...s/middleware/repo.go:321 func1()] [E] GetTags: chdir /home/git/gogs-repositories/organisation/website.git: no such file or directory
[Macaron] Completed /Organisation/website 500 Internal Server Error in 9.814868ms
</pre>
when looking in my directory structure, the website.git is available...
<pre>
$ ls -l /data/docker/gogs/git/gogs-repositories/ictcorebiz/website.git/
total 36
-rw-r--r--   1 xyzuser xyzgroup   23 Sep 30 21:38 HEAD
drwxr-xr-x   2 xyzuser xyzgroup 4096 Sep 30 21:38 branches
-rw-r--r--   1 xyzuser xyzgroup   66 Sep 30 21:38 config
-rw-r--r--   1 xyzuser xyzgroup   73 Sep 30 21:38 description
drwxr-xr-x   2 xyzuser xyzgroup 4096 Sep 30 21:38 hooks
drwxr-xr-x   2 xyzuser xyzgroup 4096 Oct 11 18:57 info
drwxr-xr-x 112 xyzuser xyzgroup 4096 Oct 11 18:57 objects
-rw-r--r--   1 xyzuser xyzgroup   85 Sep 30 23:48 packed-refs
drwxr-xr-x   4 xyzuser xyzgroup 4096 Sep 30 21:38 refs
</pre>

**Comment by Unknwon** [comment]
Thanks your feedback!

cc @0rax I think it may because the default location for storing repositories has been changed?

**Comment by ictcorebiz** [comment]
when going back to 0.6.15 all works fine :) so I can at least continue my development.

<code>
docker run --restart always --detach --name=gogs -p 10022:22 -p 10080:3000 -v /data/docker/gogs:/data gogs/gogs:0.6.15
</code>

**Comment by Unknwon** [comment]
@ictcorebiz yes, I've cc-ed maintainer of it. 

@0rax  https://github.com/gogits/gogs/blob/develop/docker%2Fbuild.sh#L24 can here make s symbolic link from old location `home/git` to keep backward compatibility?

**Comment by 0rax** [comment]
This should fix it.

Old app config makes gogs try to get git repositories from /home/git but
now git home is on /data/git.

Can you dump your `app.ini` here, please. So that I can make sure that this
will fix it.

**Comment by ictcorebiz** [comment]
probably you mean the section:
<pre>
[repository]
ROOT = /home/git/gogs-repositories
</pre>

**Comment by 0rax** [comment]
Yep just needed this one. Will create backward compatibility fix right now.

**Comment by 0rax** [comment]
@ictcorebiz my commit bfed40e should fix it.

If you can try it yourself, i've build an image for this commit available on docker hub as `orax/gogs:backward-compat`.

**Comment by ictcorebiz** [comment]
GREAT! works!
Can I continue using this image, or does it contain unfinished other work?

**Comment by 0rax** [comment]
This image should now be pretty much done, we just took another approach to the way the container works (we reduced its size drastically). Consider just switching from `orax/gogs:backward-compat` to the `gogs` one when my PR will be merged.

For more insight on the bug, I just totally forgot about the fact that the path of the git repository was stored as a full path and not a relative path to `git` user home.

Just checked everything, and this docker image should be pretty much backward compatible and should not contain any more bug.

My thought on this is that we are pretty much production ready as of right now, but it might still contains some minor bugs that I will try to take care of them ASAP if they appears.

That's just my own thoughts, so it is pretty much up to you. I don't consider it bleeding edge myself.
